packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -184,6 +184,60 @@ class VisualStudio {
     }
   }
 
+  /// The path to cl.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get clPath {
+    return _getMsvcBinPath('cl.exe');
+  }
+
+  /// The path to lib.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get libPath {
+    return _getMsvcBinPath('lib.exe');
+  }
+
+  /// The path to link.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get linkPath {
+    return _getMsvcBinPath('link.exe');
+  }
+
+  String? _getMsvcBinPath(String executable) {
+    final VswhereDetails? details = _bestVisualStudioDetails;
+    if (details == null || !details.isUsable || details.installationPath == null || details.msvcVersion == null) {
+      return null;
+    }
+
+    return _fileSystem.path.joinAll(<String>[
+      details.installationPath!,
+      'VC',
+      'Tools',
+      'MSVC',
+      details.msvcVersion!,
+      'bin',
+      'Hostx64',
+      'x64',
```

Comment:
Are these paths correct on an arm64 machine, or are there native versions?

Reply:
Do we support arm64?

It doesn't look like it: https://github.com/flutter/flutter/issues/53120

Reply:
I believe the host support has landed. Presumably at least the `Hostx64` part of this is the arch of the binary itself, rather than what the target arch is?

Reply:
@loic-sharma Would you be able to help here?

Reply:
My Arm64 machine has the following toolchains:

```
VC\Tools\MSVC\<msvc version>\bin\Hostarm64\arm64
VC\Tools\MSVC\<msvc version>\bin\Hostarm64\x64
VC\Tools\MSVC\<msvc version>\bin\Hostx64\x64
VC\Tools\MSVC\<msvc version>\bin\Hostx64\arm64
```

Currently we build x64 executables on Windows Arm64 hosts. These executables run using x64 emulation. We don't support creating Arm64 executables yet, nor do we support cross-compilation. For now, using the x64 toolchain on Arm64 hosts seems reasonable. /cc @pbo-linaro who is working on Windows Arm64 support.

Reply:
I ran `test/general.shard/windows/native_assets_test.dart` on my Windows Arm64 machine, all 11 tests passed.

Reply:
I think it's fine to use x64 (hostx64) cl.exe running under emulation, it's always avaiable, and avoid special case regarding host machine.

Reply:
Won't that unnecessarily slow the build process?

Reply:
On the M1 Macs the x64 emulation is pretty fast. I have no idea about the quality of x64 emulation on Windows machines.

It's probably cleaner to use the `Hostarm64` if it is available. Though, maybe it's better to do that in a follow up PR by someone who can actually test it, as I can't.